lib/BaseDownloadController.js

```diff
@@ -195,6 +196,19 @@ module.exports = class BaseDownloadController extends CompositeController {
         }
       }
     }
+
+    let reqopt = new FetchEnvs(this);
+    let headersFrom = objectPath.get(requestOptions, 'headersFrom');
+    if (headersFrom) {
+      let headersFromTemp = await reqopt.processEnvFrom(headersFrom);
+      let headers = objectPath.get(requestOptions, 'headers');
+      for (const header of headersFromTemp) {
+        const data = header?.data;
+        headers = { ...headers, ...data };
+      }
+      requestOptions = { ...requestOptions, headers };
+    }
```

Comment:
- good to use const instead of let where possible.
- headers was already defined above, use a new name to be clear what the intention is.
- note: the below suggestion is untested, so make sure it works before just accepting any changes.

```suggestion
    const reqopt = new FetchEnvs(this);
    const headersFrom = objectPath.get(requestOptions, 'headersFrom');
    if (headersFrom) {
      const headersFromTemp = await reqopt.processEnvFrom(headersFrom);
      let mergedHeaders = { ...headers };
      for (const header of headersFromTemp) {
        const data = header?.data;
        mergedHeaders = { ...mergedHeaders, ...data };
      }
      requestOptions = { ...requestOptions, headers: mergedHeaders };
    }
```